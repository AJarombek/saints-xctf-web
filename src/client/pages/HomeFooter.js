/**
 * Footer section of the home page linking to different sources.
 * @author Andrew Jarombek
 * @since 1/21/2020
 */

import React from 'react';

const HomeFooter = () => {
    return (
        <div className="sxctf-home-footer">
          <a href="mailto:andrew@jarombek.com">Contact Us</a>
        </div>
    );
};

export default HomeFooter;
