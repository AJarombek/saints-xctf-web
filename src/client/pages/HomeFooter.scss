/*
 * Sass styling for the HomeFooter component
 * @author Andrew Jarombek
 * @since 1/22/2020
 */

.sxctf-home-footer {

}
