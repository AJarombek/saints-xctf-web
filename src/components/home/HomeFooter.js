/**
 * Footer section of the home page linking to different sources.
 * @author Andrew Jarombek
 * @since 1/21/2020
 */

import React from 'react';
import saints_xctf_logo from '../../../assets/saintsxctf_logo.png';

const HomeFooter = () => {
  return (
    <div className="sxctf-home-footer">
      <a className="contact-us-link" href="mailto:andrew@jarombek.com">Contact Us</a>
      <a className="sxctf-logo-link" href="#">
        <figure className="sxctf-logo">
          <img src={saints_xctf_logo}/>
        </figure>
      </a>
    </div>
  );
};

export default HomeFooter;
