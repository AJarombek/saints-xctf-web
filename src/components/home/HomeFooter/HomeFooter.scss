/*
 * Sass styling for the HomeFooter component
 * @author Andrew Jarombek
 * @since 1/22/2020
 */

@import "../../../vars";
@import "../../../fonts";

.sxctf-home-footer {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #fdfdfd;
  padding: 20px 0;

  .contact-us-link {
    @include roboto-slab-bold;
    text-decoration: none;
    color: #777;
    transition: color 0.5s ease;

    &:hover {
      color: $saints-xctf-red;
      text-decoration: underline;
    }
  }

  .sxctf-logo-link {
    figure {
      height: 50px;
      width: 50px;
      cursor: pointer;

      img {
        height: 100%;
        width: 100%;
        filter: grayscale(1) brightness(1.3);
        transition: filter 0.5s ease;

        &:hover {
          filter: none;
        }
      }
    }
  }
}
