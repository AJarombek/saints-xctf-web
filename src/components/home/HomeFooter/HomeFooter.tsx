/**
 * Footer section of the home page linking to different sources.
 * @author Andrew Jarombek
 * @since 1/21/2020
 */

import React from 'react';

// @ts-ignore
import saints_xctf_logo from '../../../../assets/saintsxctf_logo.png';

interface IProps {
    showContactUs?: boolean;
}

const HomeFooter: React.FunctionComponent<IProps> = ({ showContactUs = true }) => {
  return (
    <div className="sxctf-home-footer">
        {showContactUs && <a className="contact-us-link" href="mailto:andrew@jarombek.com">Contact Us</a>}
      <a className="sxctf-logo-link" href="#">
        <figure className="sxctf-logo">
          <img src={saints_xctf_logo} alt=""/>
        </figure>
      </a>
    </div>
  );
};

export default HomeFooter;
