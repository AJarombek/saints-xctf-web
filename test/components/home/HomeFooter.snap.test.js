/**
 * Snapshot test for the HomeFooter component.
 * @author Andrew Jarombek
 * @since 2/23/2020
 */

import React from 'react';
import HomeFooter from '../../../src/components/home/HomeFooter';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<HomeFooter />).toJSON();
  expect(tree).toMatchSnapshot();
});
