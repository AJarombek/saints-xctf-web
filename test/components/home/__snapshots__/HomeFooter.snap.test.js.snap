// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sxctf-home-footer"
>
  <a
    className="contact-us-link"
    href="mailto:andrew@jarombek.com"
  >
    Contact Us
  </a>
  <a
    className="sxctf-logo-link"
    href="#"
  >
    <figure
      className="sxctf-logo"
    >
      <img
        src="test-file-stub"
      />
    </figure>
  </a>
</div>
`;
